### PR TITLE
Add images for cilium v1.9.4 [new images]

### PR DESCRIPTION
**Pull request by jcaamano**
Signed-off-by: Jaime Caamaño Ruiz <jcaamano@suse.com>

#### Pull Request Checklist ####

- [x] Change does not remove any existing Images or Tags in index-list.txt
- [x] Change does not remove / overwrite exiting Images or Tags in Rancher DockerHub
- [x] New entries are in format `SOURCE DESTINATION TAG`
- [x] New entries are added to the correct section of the list (sorted lexographically)
- [ ] New entries have a repo created in Rancher Dockerhub (where the image will be mirrored to)
- [x] Changes to scripting or CI config have been tested to the best of your ability

#### Types of Change ####

New images: cilium v1.9.4 (latest released version)

#### Linked Issues ####
https://github.com/rancher/rancher/issues/15087
https://github.com/rancher/rke2/issues/490

### CODE REVIEW

File: images-list
Comment (cbron) [inline images-list:117]: * @brandond can skopeo script handle image sha's ? It looks like it, but I don't think we have tried before. 
* None of these are created in dockerhub, you can DM me on how to get those created. But that is one of the checkboxes that needs to be checked before this can be merged. 
* Need a real issue created somewhere for purpose of change. You can put this in rancher/rancher if you don't have a place.

**Review by cbron — CHANGES_REQUESTED** [review]

File: images-list
Comment (brandond) [reply images-list:117]: @cbron I don't think it will handle a sha256 properly, but I can test out-of-band if you want to wait a bit...

Comment (brandond) [reply images-list:117]: Oh I see, this isn't actually a sha256 blob hash, it's just a tag that corresponds to a git commit hash. That's fine.

**Review by brandond — DISMISSED** [review]
LGTM once the repos are created.

**Review by erikwilson — APPROVED** [review]

File: images-list
Comment (cbron) [inline images-list:113]: This does not exist in dockerhub: https://hub.docker.com/r/rancher/cilium-cilium/

**Review by cbron — CHANGES_REQUESTED** [review]

File: images-list
Comment (cbron) [reply images-list:113]: @brandond Think we've talked about this before but it would be cool if during CI we did a skopeo inspect on the repo to make sure it exists, and then a 2nd inspect on the tag to make sure it doesn't exist. I will add that as tech debt if you agree. Seems like it would be easy.

Comment (jcaamano) [reply images-list:113]: Pfff my bad, created a new issue: https://github.com/rancherlabs/eio/issues/362

Comment (brandond) [reply images-list:113]: Yeah, I have been meaning to add some PR checks. I think we'll need to use the docker API to poke at the repo if its empty, but that's not too difficult.

File: images-list
Comment (cbron) [inline images-list:128]: Why did these get deleted ? Nothing should get deleted from this file.

File: images-list
Comment (cbron) [inline images-list:122]: These look good now.

**Review by cbron — CHANGES_REQUESTED** [review]

File: images-list
Comment (jcaamano) [reply images-list:128]: hmmm bad rebase? Fixed!

**Review by cbron — APPROVED** [review]

**Pull request merged**
